com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs

```diff
@@ -0,0 +1,363 @@
+limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System;
+using System.IO;
+using System.Linq;
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+using UnityEngine;
+
+namespace RosSharp.Urdf
+{
+    public static class UrdfRobotExtensions
+    {
+        static string tagName = "robot";
+        static string collisionObjectName = "Collisions";
+        public static ImportSettings importsettings;
+
+        public static void Create()
+        {
+            CreateTag();
+            GameObject robotGameObject = new GameObject("Robot");
+
+            robotGameObject.tag = tagName;
+            robotGameObject.AddComponent<UrdfRobot>();
+            robotGameObject.AddComponent<RosSharp.Control.Controller>();
+
+            UrdfPlugins.Create(robotGameObject.transform);
+
+            UrdfLink urdfLink = UrdfLinkExtensions.Create(robotGameObject.transform).GetComponent<UrdfLink>();
+            urdfLink.name = "base_link";
+            urdfLink.IsBaseLink = true;
+        }
+
+        #region Import
+
+        // Note: The Create() function is now broken into multiple pipeline stages to facilitate having
+        // both a non-blocking coroutine Create function as well as a blocking synchronous one needed for stable runtime import.
+
+        // Is used to pass data and parameters between stages of the import pipeline.
+        private class ImportPipelineData
+        {
+            public ImportSettings settings;
+            public bool wasRuntimeMode;
+            public bool loadStatus;
+            public bool forceRuntimeMode;
+            public Robot robot;
+            public GameObject robotGameObject;
+            public Stack<Tuple<Link, Transform, Joint>> importStack;
+        }
+
+        // Initializes import pipeline and reads the urdf file.
+        private static ImportPipelineData ImportPipelineInit(string filename, ImportSettings settings, bool loadStatus, bool forceRuntimeMode)
+        {
+            ImportPipelineData im = new ImportPipelineData();
+            im.settings = settings;
+            im.loadStatus = loadStatus;
+            im.wasRuntimeMode = RuntimeURDF.IsRuntimeMode();
+            im.forceRuntimeMode = forceRuntimeMode;
+
+            if (forceRuntimeMode) 
+            {
+                RuntimeURDF.SetRuntimeMode(true);
+            }
+
+            im.robot = new Robot(filename);
+
+            if (!UrdfAssetPathHandler.IsValidAssetPath(im.robot.filename))
+            {
+                Debug.LogError("URDF file and ressources must be placed in Assets Folder:\n" + Application.dataPath);
+                if (forceRuntimeMode) 
+                { // set runtime mode back to what it was
+                    RuntimeURDF.SetRuntimeMode(im.wasRuntimeMode);
+                }
+                return null;
+            }
+            return im;
+        }
+
+        // Creates the robot game object.
+        private static void ImportPipelineCreateObject(ImportPipelineData im) 
```

Comment:
Do we add the `ArticulationBody` component when creating the object?

Reply:
Yes, the `ArticulationBody` is added in the `UrdfInertial.Create()`, which itself is called in `UrdfLinkExtensions.Create()`